python/lsst/ci/builder/ciBuilderLib.py

```diff
@@ -260,12 +260,19 @@ def _getAllTags(self) -> Iterable[str]:
 
     def _clean(self):
         """Resets the RunDir to a state before any commands were executed
+
+        By default this just removes the RunDir from the filesystem but the
+        method serves as a hook such that any children can specify alternative
+        behavior.
         """
-        self._runAndTrap(("reset", "--hard", "init"), "There was an issue cleaning the state: {}")
-        allTags = set(self._getAllTags())
-        allTags.remove("init")
-        self._runAndTrap(("tag", "-d") + tuple(allTags), "There was an issue cleaning the state: {}")
-        sys.exit(0)
+        if os.path.isdir(self.RunDir):
+            try:
+                _log.info(f"Removing {self.RunDir}")
+                shutil.rmtree(self.RunDir)
```

Comment:
If this can ever run in parallel, you may want `ignore_errors=True` in case of leftover `.nfs*` files.